CacheManager.StackExchange.Redis.SelfCleaning.Tests/SelfCleaningRedisIntegrationTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using Moq;
+using NUnit.Framework;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning.Tests
+{
+    [TestFixture]
+    public class SelfCleaningRedisIntegrationTests
+    {
+        private const double TIME_TO_LIVE_MILLISECONDS_DIFFERENCE_THRESHOLD = 100;
+
+        private Mock<IServer> _serverMock;
+        private Mock<IDatabase> _databaseMock;
+        private Mock<IConnectionMultiplexer> _connectionMock;
+
+        private IDictionary<RedisKey, (RedisValue[] Values, DateTime InsertionTime)> _fauxDatabase;
+
+        private TimeSpan _timeToLive;
+        private ICacheManager<DummyModel> _cache;
+
+        private ICollection<(CacheItemRemovedEventArgs Args, DateTime Time)> _onRemoveByHandleInvocations;
+
+        [OneTimeSetUp]
```

Comment:
As we spoke, try to test whether the tests pass when using usual SetUp (and TearDown). If it does make problems, leave it that way.

Reply:
Surprisingly enough, it does work. However, there is one obstruction - I need the clean the configurations dictionary so that the same configuration won't be used twice in different tests (that could mean using an already disposed object, for example). Since there's no public access for clearing the `RedisConfigurations` dictionary, I need to use reflection.